sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/CertificateOperationTests.cs

```diff
@@ -115,6 +116,7 @@ public async Task UpdateStatusEventuallyCompleted()
 
             CertificateClient client = CreateClient(transport);
             CertificateOperation operation = await client.StartCreateCertificateAsync(CertificateName, s_policy);
+            operation = InstrumentMockOperation(operation);
```

Comment:
I think we should investigate why the operation is not automatically instrumented here.  For instance any method returning an [Operation like this](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/tests/Scenario/ResourceGroupCollectionTests.cs#L150) gets automatically instrumented by the test framework.  I believe the code that does that[ is here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/Instrumentation/InstrumentResultInterceptor.cs#L61) but for some reason it isn't catching this case and we should probably update it so that we don't have to manually find all of these.

Reply:
Because my test class only extends `ClientTestBase` and operations only get instrumented by `RecordedTestBase`:

https://github.com/Azure/azure-sdk-for-net/blob/b732adadb71dadfee5c083165b90e1d6a3148ffb/sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs#L352-L353

The `ClientTestBase.InstrumentOperation` merely returns the operation unchanged. As noted in my [reply below](https://github.com/Azure/azure-sdk-for-net/pull/27399#discussion_r821914946) I'm not keen on regressing any of the many track 2 libraries might be inadvertently taking advantage of this fact, so I made a new method anyone needing to mock operations without using the recording infrastructure - which is a non-starter for this test: I can't control what the service does but want to test a particular chain of responses.

/cc @JoshLove-msft

Reply:
According to these [search results](https://cs.github.com/Azure/azure-sdk-for-net?q=InstrumentOperation+ClientTestBase), outside of these related base classes, I don't see any. Everything else seems to extend from `RecordedTestBase`, including some even more-derived classes in Azure.ResourceManager.Network.